src/android/SplashScreen.java

```diff
@@ -151,13 +193,40 @@ public Object onMessage(String id, Object data) {
         }
         return null;
     }
+    
+    @Override
+    public void onConfigurationChanged(Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+
+        // Reload splash drawable when orientation changes if so configured.
+        if (newConfig.orientation != orientation) {
+            orientation = newConfig.orientation;
+            reloadDrawable();
+        }
+    }
 
-    private void removeSplashScreen() {
+    private void reloadDrawable() {
+        if (isReloadOnOrientationChange() && splashImageView != null) {
```

Comment:
Not sure if you answered in the other PR, but why not just always reload on configuration changes? It'll just no-op if there's no new image.